### PR TITLE
Analysis of the program

### PR DESCRIPTION
**Pull request by eklopfstein**
The program boots up very nicely and puts the users right into the app ready to go. The parts that have been finished look good and user friendly. The unit tests are very well done but I saw some edge cases unaccounted for.

Was the program available in UC Github on time?

Yes, I was given access to the repo immediately.

Is the program documented/commented well enough for you to understand?

The program is well organized and documented very well. It was very easy for me to understand what it was doing.

Does the program compile?

Yes, the app launched with no fatal errors.

Rationale behind your changes.

While going through the app I noticed that the unit cases do not check for the creation of bad ListItems. I added the tests to do so and then added the verification in the ListItems class to throw errors when a bad entry is given. I also removed a few spots where old code was commented out. It is best to use version control to keep unused or bad code out of the master as apposed to comments.

Three Technical Concepts I learned

I learned more about recylerviews and how to use them. They get tricky when adding and removing things and they handle it well.
I learned about adding items to firebase which is helpful as my group is currently having issues with ours.
I learned more about adding LiveData to a feed which is helpful as that is next on my group's TODO list.

### CODE REVIEW

File: app/src/main/java/edu/uc/groupProject/topten/MainActivity.kt [hunk 1]
```diff
@@ -32,20 +32,6 @@ class MainActivity : AppCompatActivity() {
...
-
-        mvm.list.observeForever {
-            it.forEach {
-                textView.append(it.title)
-                textView.append("\n")
-            }
-        }
-
-        */
-
```
Comment (eklopfstein) [inline app/src/main/java/edu/uc/groupProject/topten/MainActivity.kt:48]: Best to use version control to get rid of unused code instead of comments

File: app/src/main/java/edu/uc/groupProject/topten/dto/ListItem.kt [hunk 2]
```diff
@@ -8,7 +8,23 @@ package edu.uc.groupProject.topten.dto
...
+    init {
+        // Ensure the title is not blank
+        if (title.isBlank()) {
+            throw IllegalArgumentException("Name required")
+        }
+        // Ensure the votes are not negative
+        if (totalVotes < 0) {
+            throw IllegalArgumentException("Votes must be non-negative")
+        }
+    }
```
Comment (eklopfstein) [inline app/src/main/java/edu/uc/groupProject/topten/dto/ListItem.kt:27]: Added an init where I verified the name and vote counts are valid inputs

File: app/src/test/java/edu/uc/groupProject/topten/ListItemDataUnitTest.kt [hunk 3]
```diff
@@ -93,4 +90,24 @@ class ListItemDataUnitTest {
         toStringTest = ListItem(0, "Avengers", "A movie about Batman", 100)
         assertTrue(toStringTest.toString() == "Avengers")
     }
+
+    @Test
+    fun ensureCorrectVoteCounts() {
```
Comment (eklopfstein) [inline app/src/test/java/edu/uc/groupProject/topten/ListItemDataUnitTest.kt:95]: Added test for edge case of ListItems with bad vote counts

File: app/src/test/java/edu/uc/groupProject/topten/ListItemDataUnitTest.kt [hunk 4]
```diff
@@ -93,4 +90,24 @@ class ListItemDataUnitTest {
...
+        // Code should ensure the counts are correct before they are added
+        try {
+            badListItems = ListItem(0, "Avengers", "A movie about Batman", -100)
+        } catch (ex: Exception) {
+            assertTrue(ex.message == "Votes must be non-negative")
+        }
+    }
+
+    @Test
+    fun ensureMovieHasName() {
```
Comment (eklopfstein) [inline app/src/test/java/edu/uc/groupProject/topten/ListItemDataUnitTest.kt:105]: Added test for edge case of ListItems with no name

File: app/src/main/java/edu/uc/groupProject/topten/MainActivity.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/groupProject/topten/MainActivity.kt:48]: Yes.

 Committing commented out code is considered bad practice in version control, because version control offers better alternatives:
1) Use a feature branch for experimental features.
2) Delete what is not needed.  You can always compare with history to see what used to be there.
Commented out code is considered clutter in a version control system.

File: app/src/main/java/edu/uc/groupProject/topten/dto/ListItem.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/groupProject/topten/dto/ListItem.kt:27]: Hmmm.  Good technique.

File: app/src/main/java/edu/uc/groupProject/topten/ui/main/MainFragment.kt [hunk 5]
```diff
@@ -82,44 +78,20 @@ class MainFragment : Fragment() {
...
-            "A list of favorite cereals",
-            false,
-            "Food",
-            Date(),
-            Date()
-        )
-        list.listItems = listItemsToAdd
-        viewModel.firestoreService.writeListToDatabase(list)
-
-      */
```
Comment (discospiff) [inline app/src/main/java/edu/uc/groupProject/topten/ui/main/MainFragment.kt:121]: Good.  Ditto above.  Remove commented out code from version control.

File: app/src/test/java/edu/uc/groupProject/topten/ListItemDataUnitTest.kt (hunk 3 above)
Comment (discospiff) [reply app/src/test/java/edu/uc/groupProject/topten/ListItemDataUnitTest.kt:95]: Good.  Increase test coverage.

**Comment by discospiff** [comment]
Several good suggestions here.  I recommend the group examine the suggestions closely, and they will likely find the changes worth merging.
